deploy-manage/upgrade/deployment-or-cluster/upgrade-717.md

```diff
@@ -1,7 +1,501 @@
+
+  ::::{dropdown} 7.17 {{stack}} ingest components
+  * {{beats}}: [Upgrade instructions](https://www.elastic.co/guide/en/beats/libbeat/7.17/upgrading.html)
+  * {{ls}}: [Upgrade instructions](https://www.elastic.co/guide/en/logstash/7.17/upgrading-logstash.html)
+  * {{fleet}} managed {{agent}}: [Upgrade instructions](https://www.elastic.co/guide/en/fleet/7.17/upgrade-elastic-agent.html)
+  * Standalone {{agent}}: [Upgrade instructions](https://www.elastic.co/guide/en/fleet/7.17/upgrade-standalone.html)
+  * Elastic APM: [Upgrade instructions](https://www.elastic.co/guide/en/apm/guide/7.17/upgrade.html)
+  ::::
+
+  After upgrading the cluster to 8.19.x and before proceeding to {{version.stack}}, upgrade all ingest components to 8.19.x. This step will be covered later in this guide.
+  
+* **Client libraries:**
+
+  If you use custom-developed applications that rely on [{{es}} client libraries](/reference/elasticsearch-clients/index.md), make sure to include them in your plan. Client libraries must be upgraded after each major {{es}} upgrade to ensure support and compatibility.
+
+  Applications that use deprecated or removed APIs might require code updates, or you can use the [REST API compatibility](elasticsearch://reference/elasticsearch/rest-apis/compatibility.md) feature to maintain compatibility with the next major version.
+
+  This topic will be revisited in more detail during the preparation steps for each major upgrade.
+
+* **Orchestration platforms:**
+
+  * **ECK**: If you are running an ECK version earlier than 3.x, you need to [upgrade ECK to 3.x](/deploy-manage/upgrade/orchestrator/upgrade-cloud-on-k8s.md) before the final upgrade to 9.x. This can be done either at the beginning, before the initial upgrade, or between the two upgrade phases.
+
+  * **ECE**: If you are running an ECE version earlier than 4.x, you need to [upgrade your ECE platform to 4.x](/deploy-manage/upgrade/orchestrator/upgrade-cloud-enterprise.md) before the final upgrade to 9.x. This upgrade must be performed after upgrading your deployments to 8.19.x, because ECE 4.x is not compatible with 7.x deployments.
+
+Finally, we strongly recommend [testing the full upgrade process in a non-production environment](/deploy-manage/upgrade/plan-upgrade.md#test-in-a-non-production-environment) before applying it to production.
+
+## Upgrade Step 1: 7.17.x → 8.19.x
+
+This step covers upgrading your deployment from 7.17.x to 8.19.x, following the [planning phase](#planning) and assuming that all ingest components and client libraries are compatible with 8.19.x.
+
+It's highly recommended to start this upgrade from the latest 7.17.x patch release to ensure that you’re using the most recent version of the **Upgrade Assistant**. You should also upgrade to the latest available 8.19.x patch release so that the same benefits apply when you later upgrade to 9.x.
```

Comment:
this bolding doesn't add a lot of value imo
```suggestion
It's highly recommended to start this upgrade from the latest 7.17.x patch release to ensure that you’re using the most recent version of the Elastic Upgrade Assistant. You should also upgrade to the latest available 8.19.x patch release so that the same benefits apply when you later upgrade to 9.x.
```